conformance/src/main/scala/za/co/absa/enceladus/conformance/DynamicConformanceJob.scala

```diff
@@ -173,9 +143,55 @@ object DynamicConformanceJob {
     spark
   }
 
+  private def inferVersion(hdfsPublishPath: String, reportDate: String)
+                          (implicit fsUtils: FileSystemVersionUtils):Int = {
+    val newVersion = fsUtils.getLatestVersion(hdfsPublishPath, reportDate) + 1
+    log.warn(s"Report version not provided, inferred report version: $newVersion")
+    log.warn("This is an EXPERIMENTAL feature.")
+    log.warn(" -> It can lead to issues when running multiple jobs on a dataset concurrently.")
+    log.warn(" -> It may not work as desired when there are gaps in the versions of the data being landed.")
+    newVersion
+  }
+
+  private def initFunctionalExtensions()(implicit spark: SparkSession): Unit = {
+    import za.co.absa.spline.core.SparkLineageInitializer._ // Enable Spline
+    spark.enableLineageTracking()
+
+    import za.co.absa.atum.AtumImplicits.SparkSessionWrapper
+    spark.enableControlMeasuresTracking().setControlMeasuresWorkflow("Conformance")
+    Atum.setAllowUnpersistOldDatasets(true) // Enable control framework performance optimization for pipeline-like jobs
+    MenasPlugin.enableMenas() // Enable Menas
+  }
+
+  private def initPerformanceMeasurer(stdPath: String)
+                                     (implicit spark: SparkSession, fsUtils: FileSystemVersionUtils): PerformanceMeasurer = {
+    // init performance measurer
+    val performance = new PerformanceMeasurer(spark.sparkContext.appName)
+    val stdDirSize = fsUtils.getDirectorySize(stdPath)
+    performance.startMeasurement(stdDirSize)
+    performance
+  }
+
+  private def performTheConformance(conformance: Dataset, inputData: sql.Dataset[Row], enableCF: Boolean)
+                                   (implicit spark: SparkSession, cmd: CmdConfig, fsUtils: FileSystemVersionUtils, dao: EnceladusDAO): DataFrame = {
+    try {
+      DynamicInterpreter.interpret(conformance, inputData, isExperimentalRuleEnabled(), isCatalystWorkaroundEnabled(), enableCF)
+    } catch {
+      case e: ValidationException =>
+        AtumImplicits.SparkSessionWrapper(spark).setControlMeasurementError("Conformance", e.getMessage, e.techDetails)
+        throw e
+      case NonFatal(e) =>
+        val sw = new StringWriter
+        e.printStackTrace(new PrintWriter(sw))
+        AtumImplicits.SparkSessionWrapper(spark).setControlMeasurementError("Conformance", e.getMessage, sw.toString)
+        throw e
+    }
+  }
+
   private def processResult(result: DataFrame, performance: PerformanceMeasurer, publishPath: String, stdPath: String, reportVersion: Int)
                            (implicit spark: SparkSession, cmd: CmdConfig, fsUtils: FileSystemVersionUtils): Unit = {
-    val withPartCols = result.withColumn(infoDateColumn, lit(new java.sql.Date(format.parse(cmd.reportDate).getTime)))
+    val withPartCols = result
+      .withColumn(infoDateColumn, to_date(lit(cmd.reportDate), reportDateFormat))
```

Comment:
```
      to_date(lit(cmd.reportDate), reportDateFormat)
```
This is the only non-refactoring change in this file, right?

Reply:
Nope, the changes in _OozieRepository_ are also "functional"

Reply:
I mean everything else in **this file** is refactoring

Reply:
Eh, sorry for misunderstanding. Yes, only in this file.